src/main/java/christmas/domain/EventProcess.java

```diff
@@ -0,0 +1,100 @@
+package christmas.domain;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class EventProcess {
+    private static final int MINIMUM_TOTAL_ORDER_PRICE_FOR_GIFT = 120000;
+    private static final int MINIMUM_TOTAL_ORDER_PRICE_FOR_ALL_EVENT = 10000;
+    private static final int D_DAY_DISCOUNT_AMOUNT_PER_DAY = 100;
+    private static final int D_DAY_DISCOUNT_DEFAULT_AMOUNT = 1000;
+    private static final int FIRST_DAY_NUMBER = 1;
+    private static final int WEEKDAY_DESSERT_DISCOUNT_AMOUNT = 2023;
+    private static final int WEEKEND_MAIN_DISCOUNT_AMOUNT = 2023;
+    private static final int SPECIAL_DISCOUNT_AMOUNT = 1000;
+
+    public EventResult takeAllBenefit(Date date, Order order) {
+        Map<BenefitType, Integer> allBenefit = new HashMap<>();
+        allBenefit.put(BenefitType.GIFT, takeGift(order));
+        allBenefit.put(BenefitType.D_DAY, takeDDayDiscount(date));
+        allBenefit.put(BenefitType.WEEKDAY, takeWeekdayDiscount(date, order));
+        allBenefit.put(BenefitType.WEEKEND, takeWeekendDiscount(date, order));
+        allBenefit.put(BenefitType.SPECIAL, takeSpecialDiscount(date));
+
+        cancelAllBenefitIfMeetCondition(order, allBenefit);
+
+        return new EventResult(allBenefit);
```

Comment:
(개인적인 생각입니다!)

```suggestion
          return new EventResult(new HashMap<>() {{
            this.put(BenefitType.GIFT, takeGift(order));
            this.put(BenefitType.D_DAY, takeDDayDiscount(date));
            this.put(BenefitType.WEEKDAY, takeWeekdayDiscount(date, order));
            this.put(BenefitType.WEEKEND, takeWeekendDiscount(date, order));
            this.put(BenefitType.SPECIAL, takeSpecialDiscount(date));
        }});
```
이런식으로 생성과 동시에 HashMap의 원소를 초기화 하는 방법도 있어요 :)

Reply:
훨씬 깔끔하네요,, 감사합니다!!